python/mlc_chat/protocol/protocol_utils.py

```diff
@@ -25,13 +26,33 @@ def get_unsupported_fields(request: RequestProtocol) -> List[str]:
     """Get the unsupported fields of the request.
     Return the list of unsupported field names.
     """
-    if isinstance(request, OpenAIRequestProtocol):
+    if isinstance(request, (OpenAICompletionRequest, OpenAIChatCompletionRequest)):
         return openai_api_get_unsupported_fields(request)
     raise RuntimeError("Cannot reach here")
 
 
-def get_generation_config(request: RequestProtocol) -> GenerationConfig:
+def get_generation_config(
+    request: RequestProtocol,
+    extra_stop_tokens: Optional[List[int]] = None,
+    extra_stop_str: Optional[List[str]] = None,
+) -> GenerationConfig:
     """Create the generation config in MLC LLM out from the input request protocol."""
-    if isinstance(request, OpenAIRequestProtocol):
-        return openai_api_get_generation_config(request)
-    raise RuntimeError("Cannot reach here")
+    kwargs: Dict[str, Any]
+    if isinstance(request, (OpenAICompletionRequest, OpenAIChatCompletionRequest)):
+        kwargs = openai_api_get_generation_config(request)
```

Comment:
Just curious, what's the difference between `openai_api_get_generation_config` and `get_generation_config`?

Reply:
We may not only have OpenAI protocols. Say we have OpenAI protocols and our own protocols at the same time. Both protocols are defined in their respective files (like `python/mlc_chat/protocol/openai_api_protocol.py`), where their own conversion rules (like `openai_api_get_generation_config`) are implemented.

`get_generation_config` dispatches request to different protocol conversion rules.